src/drawing/DrawingAnnotations.tsx

```diff
@@ -29,23 +35,45 @@ const DrawingAnnotations = (props: Props): JSX.Element => {
         drawnPathGroups,
         isCreating,
         location,
+        resetDrawing,
         setActiveAnnotationId,
         setDrawingLocation,
+        setReferenceId,
+        setStaged,
         setStatus,
+        status,
     } = props;
+    const [isDrawing, setIsDrawing] = React.useState<boolean>(false);
     const [stagedRootEl, setStagedRootEl] = React.useState<DrawingSVGRef | null>(null);
     const hasDrawnPathGroups = drawnPathGroups.length > 0;
-    const uuidRef = React.useRef<string>(uuid.v4());
+    const stagedGroupRef = React.useRef<SVGGElement>(null);
+    const { current: drawingSVGGroup } = stagedGroupRef;
 
     const handleAnnotationActive = (annotationId: string | null): void => {
         setActiveAnnotationId(annotationId);
     };
+    const handleDelete = (): void => {
+        resetDrawing();
+    };
+    const handleDrawingMount = (uuid: string): void => {
+        setReferenceId(uuid);
+    };
+    const handleReply = (): void => {
+        setStaged({ location, pathGroups: drawnPathGroups });
+        setStatus(CreatorStatus.staged);
+    };
     const handleStart = (): void => {
+        if (status === CreatorStatus.staged) {
```

Comment:
Can this be handled in the reducer? Should we reset the drawing when the location changes?